src/main/java/com/github/cameltooling/lsp/internal/parser/ParserXMLFileHelper.java

```diff
@@ -169,7 +169,12 @@ private Document getDocumentWithLineInformation(TextDocumentItem textDocumentIte
 	@Override
 	public CamelURIInstance createCamelURIInstance(TextDocumentItem textDocumentItem, Position position, String camelComponentUri) {
 		Node correspondingCamelNode = getCorrespondingCamelNodeForCompletion(textDocumentItem, position.getLine());
-		return new CamelURIInstance(camelComponentUri, correspondingCamelNode);
+		CamelURIInstance uriInstance = new CamelURIInstance(camelComponentUri, correspondingCamelNode);
+		uriInstance.setDocument(textDocumentItem);
```

Comment:
it seems that the initialization of the CamelURiInstance is duplicated than what is in the ParserJavaFileHelper.
It confirms that it should be done in the CamelURiInstance itself (as I mentioned in the preliminary PR)